pkg/api/v1/pod/util.go

```diff
@@ -301,6 +301,14 @@ func IsPodReady(pod *v1.Pod) bool {
 	return IsPodReadyConditionTrue(pod.Status)
 }
 
+// IsPodTerminal returns true if a pod is terminal, all containers are stopped and cannot ever regress.
```

Comment:
nit: can we name this something like IsPodComplete or IsPodDone? Terminal and Terminating sound close and might become confusing in the future.

Reply:
is to provide consistency with the kubelet terminlogy
```
grep -r PodTerminal pkg/kubelet/
pkg/kubelet/kubelet_pods.go:            if kl.isAdmittedPodTerminal(p) && !kl.podWorkers.IsPodTerminationRequested(p.UID) {
pkg/kubelet/kubelet_pods.go:// isAdmittedPodTerminal returns true if the provided config source pod is in
pkg/kubelet/kubelet_pods.go:func (kl *Kubelet) isAdmittedPodTerminal(pod *v1.Pod) bool {
pkg/kubelet/kubelet_pods.go:            if kl.isAdmittedPodTerminal(pod) {
```